### ISSUE TITLE
🐛 [firebase_messaging] v8.0.0-dev.2 Background handler is not working

### ISSUE THREAD

**Issue by lazylazyllama** [FirebaseExtended/flutterfire#3992]
## Bug report

When I receive a message in the foreground, it is working. But if my App is in the background I get this error:

```
D/FLTFireMsgReceiver(22164): broadcast received for message
W/FLTFireMsgService(22164): Attempted to start a duplicate background isolate. Returning...
W/e.gebatzens.si(22164): Accessing hidden method Lsun/misc/Unsafe;->compareAndSwapObject(Ljava/lang/Object;JLjava/lang/Object;Ljava/lang/Object;)Z (greylist, linking, allowed)
I/flutter (22164): FlutterFire Messaging: An error occurred in your background messaging handler:
I/flutter (22164): Converting object to an encodable object failed: Instance of 'RemoteMessage'
```

**Comment by ghenry** [comment]
Not this simple I presume? ` Converting object to an encodable object failed: Instance of 'RemoteMessage'`

**Comment by lazylazyllama** [comment]
Yeah, thats the issue, but it only happens when the app runs in the background for some reason 🤔

**Comment by markusaksli-nc** [comment]
I can't repro this. I just get
```
D/FLTFireMsgReceiver(23902): broadcast received for message
W/FirebaseMessaging(23902): Missing Default Notification Channel metadata in AndroidManifest. Default value will be used.
I/flutter (23902): Handling a background message 0:1604317240641514%a9039156a9039156
```
with `1.24.0-7.0.pre.51`.
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.24.0-7.0.pre.51, on Microsoft Windows [Version 10.0.19041.572], locale en-US)
    • Flutter version 1.24.0-7.0.pre.51 at C:\Development\flutter_master
    • Framework revision 734d2a22fe (2 hours ago), 2020-11-02 15:23:02 +0530
    • Engine revision 3460519398
    • Dart version 2.12.0 (build 2.12.0-3.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6858069\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.5)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.6.30320.27
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6858069
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\203.5419.21
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (5 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)           • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.572]
    • Web Server (web)            • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 86.0.4240.111
    • Edge (web)                  • edge          • web-javascript • Microsoft Edge 86.0.622.51

• No issues found!
```
</details>

Though I'm not sure if this is the expected output either. What version of flutter did you get this on? Are you doing any background message handling that would cause this?

**Comment by Ehesp** [comment]
Can you share the code in the background handler?

**Comment by lazylazyllama** [comment]
My bad. It seems like I fooled myself. I thought I tried to do a print at the very top of my background handler and it didn't work, but after trying it again it in fact did work and the issue was a `jsonEncode(message)` function call somewhere in my background handler.

Maybe it would be nice to have `RemoteMessage` implement `toJson`, so that we can convert it to json again? I would like to do this for logging purposes.
